cupy/math/misc.py

```diff
@@ -5,6 +5,46 @@
 # TODO(okuta): Implement convolve
 
 
+def normalize_axis_index(axis, ndim):
```

Comment:
You can write this way: https://github.com/cupy/cupy/blob/678466d06e237d28cc18fd483e15d71771d3f8a3/cupy/core/_routines_sorting.pyx#L83-L86

`cupy/math/misc.py` is not a correct place to implement this function.
For this PR, you can implement it in `cupy/linalg/product.py` (privately as `_normalize_axis_index`).
Or if you want to make it available to other functions, maybe `cupy/utils.pyx` is the right place, also with `_` prefix.

Other points that need fixes:
- The documentation format is different from CuPy's convention.
- Raise `_errors._AxisError` instead of `ValueError`.

Reply:
Added the fixes. Please do have a look